backendRoot/src/main/java/com/virtuslab/gitmachete/backend/root/GitMacheteRepositoryBuilder.java

```diff
@@ -241,10 +269,23 @@ private SyncToParentStatus deriveSyncToParentStatus(IGitCoreRepository gitCoreRe
     }
   }
 
-  /**
-   * @return Option of {@link IGitCoreLocalBranch} or if branch with given name doesn't exist returns empty Option
-   */
-  private Optional<IGitCoreLocalBranch> getCoreBranchFromName(IGitCoreRepository gitCoreRepository, String branchName) {
-    return Try.of(() -> Optional.of(gitCoreRepository.getLocalBranch(branchName))).getOrElse(() -> Optional.empty());
+  @RequiredArgsConstructor
+  private static class AncestorityCache {
+    private final IGitCoreRepository repository;
+    Map<Tuple2<BaseGitCoreCommit, BaseGitCoreCommit>, Boolean> cache = new HashMap<>();
```

Comment:
You mean smth that will compute the value if it is absent, right? If so it is still the case that haunts as since streams and exceptions discussion. :( 

Ofc there are:
https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#computeIfAbsent-K-java.util.function.Function-
https://www.javadoc.io/doc/io.vavr/vavr/0.9.0/io/vavr/collection/Map.html#computeIfAbsent-K-java.util.function.Function-
but I am afraid that they cannot be used (easily)

Reply:
Oh ok, I forgot again :( let's leave as is then